### PR TITLE
Align z normalization with the papers

### PR DESCRIPTION
**Pull request by borawhocodess**
Now, the bucket edges are computed after target normalization.

And during inference, we z normalize both features and targets using training statistics, compute predictions on the normalized data, then renormalize predictions to the original target scale.

Also updated docstrings.

### CODE REVIEW

File: nanotabpfn/utils.py [hunk 1]
```diff
@@ -20,20 +20,18 @@ def make_global_bucket_edges(filename, n_buckets=100, device=get_default_device(
...
+        num_tables_to_use = min(num_tables, max_y // num_datapoints)
+
+        for i in range(num_tables_to_use):
+            y_i = np.array(y[i, :], dtype=np.float32)
+            y_i_mean = np.mean(y_i)
+            y_i_std = np.std(y_i) + 1e-8
+            y_i_normalised = (y_i - y_i_mean) / y_i_std
+            ys_normalised.append(y_i_normalised)
+
+    ys_concat = np.concatenate(ys_normalised, axis=0)
```
[the remark this answers is not in the document]
Comment (AlexanderPfefferle) [reply nanotabpfn/utils.py:34]: @borawhocodess I don't think we need to do the num_tables_to_use check, but the vectorized implementation would indeed be better, could you test this one and create a commit with it (without the if statement)?

File: nanotabpfn/interface.py [hunk 2]
```diff
@@ -67,12 +69,19 @@ def get_feature_preprocessor(X: np.ndarray | pd.DataFrame) -> ColumnTransformer:
...
         transformers=[
             ('num', num_transformer, num_mask),
             ('cat', cat_transformer, cat_mask)
         ]
     )
+
+    preprocessor = Pipeline(
+        [
+            ("cols", cols),
+            ("scaler", StandardScaler()),
```
Comment (AlexanderPfefferle) [inline nanotabpfn/interface.py:82]: Why add the StandardScaler?

File: nanotabpfn/interface.py [hunk 3]
```diff
@@ -184,6 +190,7 @@ def predict(self, X_test: np.ndarray) -> np.ndarray:
             y_tensor = torch.tensor(y, dtype=torch.float32, device=self.device).unsqueeze(0)
 
             logits = self.model((X_tensor, y_tensor), single_eval_pos=len(self.X_train), num_mem_chunks=self.num_mem_chunks).squeeze(0)
-            preds = self.normalized_dist.mean(logits)
+            preds_n = self.dist.mean(logits)         
+            preds = preds_n * self.y_train_std + self.y_train_mean
```
Comment (AlexanderPfefferle) [inline nanotabpfn/interface.py:194]: Do we actually need to change anything here if we normalize the dist in utils.py?
